src/engine/renderer/tr_backend.cpp

```diff
@@ -4716,6 +4716,7 @@ static void RB_RenderView( bool depthPass )
 
 	if ( backEnd.viewParms.portalLevel > 0 )
 	{
+		if ( r_liquidMapping->integer )
 		{
```

Comment:
How this part related to liquid shading?

Reply:
`tr.portalRenderImage` is used as `u_PortalMap` in `liquid_fp.glsl`

Note that this is the part of the liquid code that is broken.